open_spiel/games/solitaire.h

```diff
@@ -0,0 +1,409 @@
+#ifndef THIRD_PARTY_OPEN_SPIEL_GAMES_SOLITAIRE_H
+#define THIRD_PARTY_OPEN_SPIEL_GAMES_SOLITAIRE_H
+
+#include <any>
+#include <array>
+#include <deque>
+#include <memory>
+#include <set>
+#include <string>
+#include <tuple>
+#include <unordered_map>
+#include <variant>
+#include <vector>
+
+#include "open_spiel/spiel.h"
+#include "open_spiel/abseil-cpp/absl/types/optional.h"
+
+// An implementation of klondike solitaire:
+// https://en.wikipedia.org/wiki/Klondike_(solitaire) More specifically, it is
+// K+ solitaire, which allows the player to play any card from the deck/waste
+// that would normally become playable after some number of draws in standard
+// klondike solitaire. For a more in-depth description of K+ solitaire, see
+// http://web.engr.oregonstate.edu/~afern/papers/solitaire.pdf. This
+// implementation also gives rewards at intermediate states like most electronic
+// versions of solitaire do, rather than only at terminal states.
+
+// ANSI color codes
```

Comment:
Do these need to be in the header file or can we put them in the .cc in an anonymous namespace?